library/core/src/any.rs

```diff
@@ -662,12 +664,39 @@ impl dyn Any + Send + Sync {
 /// While `TypeId` implements `Hash`, `PartialOrd`, and `Ord`, it is worth
 /// noting that the hashes and ordering will vary between Rust releases. Beware
 /// of relying on them inside of your code!
-#[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Debug, Hash)]
```

Comment:
Not being able to derive these makes me sad, so I'm going to ask T-libs about this in #102371

Reply:
@fee1-dead has a PR for #![derive_const] #102049